doc/api/n-api.md

```diff
@@ -42,6 +42,39 @@ for the N-API C based functions exported by Node.js. These wrappers are not
 part of N-API, nor will they be maintained as part of Node.js. One such
 example is: [node-addon-api](https://github.com/nodejs/node-addon-api).
 
+## Implications of ABI Stability
+
+Although N-API provides an ABI stability guarantee, other parts of Node.js do
+not, and any external libraries used from the addon may not. In particular,
+neither of the following Node.js APIs provides an ABI stability guarantee:
+* the Node.js C++ APIs available via any of
+    ```C++
+    #include <node_buffer.h>
+    #include <node_context_data.h>
+    #include <node_contextify.h>
+    #include <node_mutex.h>
+    #include <node_object_wrap.h>
+    #include <node_perf_common.h>
+    #include <node_platform.h>
+    #include <node_version.h>
```

Comment:
Only `node_buffer.h`, `node_version.h`, `node_object_wrap.h` and `node.h` (+ N-API) are real public API. I’ll open a PR to put the others behind `NODE_WANT_INTERNALS`.